cfgov/unprocessed/js/modules/ClearableInput.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// Required polyfills for IE9.
+if ( !Modernizr.classlist ) { require( '../modules/polyfill/class-list' ); } // eslint-disable-line no-undef, global-require, no-inline-comments, max-len
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * ClearableInput
+ * @class
+ *
+ * @classdesc Initializes a new ClearableInput molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} A ClearableInput instance.
+ */
+function ClearableInput( element ) {
+  var BASE_CLASS = 'input-contains-label';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'ClearableInput' );
+  var _inputDom = _dom.querySelector( 'input' );
+  var _clearBtnDom = _dom.querySelector( '.' + BASE_CLASS + '_after__clear' );
```

Comment:
> is this BEM format?

I think so? Here's the breakdown I see:

`input-contains-label_after__clear`
`input-contains-label` = Block
`_after` = Element
`__clear` = Modifier

> Related question, should there be a js only hook instead of using the style hook?

What're the rules around doing this again, and we were going to move to `data-js-hook` attributes instead?